wa/workloads/exoplayer/__init__.py

```diff
@@ -0,0 +1,185 @@
+                else:
+                    filepath = os.path.join(self.video_directory, self.filename)
+                if not os.path.isfile(filepath):
+                    raise WorkloadError('{} does not exist.'.format(filepath))
+                self._selected_file = filepath
+            else:
+                files = self.video_files[self.format]
+                if not files:
+                    url = DOWNLOAD_URLS[self.format]
+                    filepath = os.path.join(self.video_directory, os.path.basename(url))
+                    self.logger.debug('Downloading {}...'.format(filepath))
+                    urllib.urlretrieve(url, filepath)
+                    self._selected_file = filepath
+                else:
+                    self._selected_file = files[0]
+                    if len(files) > 1:
+                        self.logger.warn('Multiple files for 720p found. Using {}.'
+                                         .format(self._selected_file))
+                        self.logger.warn('Use "filename"parameter instead of '
+                                         '"format" to specify a different file.')
+        return self._selected_file
+
+    def init_resources(self, context):
+        ensure_directory_exists(self.video_directory)
+        self.video_files = defaultdict(list)
+        self.enum_video_files()
+        self._selected_file = None
+
+    def enum_video_files(self):
+        for filename in os.listdir(self.video_directory):
+            for format in self.parameters['format'].allowed_values:
+                if format in filename:
+                    self.video_files[format].append(os.path.join(self.video_directory, filename))
+
+    def validate(self):
+        if (self.format and self.filename) and (self.format != self.parameters['format'].default):
+            raise ConfigError('Ether format *or* filename must be specified; but not both.')
+
+    def initialize(self, context):
+        super(ExoPlayer, self).initialize(context)
+        # Grant app all permissions
+        grant_app_permissions(self.target, self.package)
+
+    def setup(self, context):
+        self.device_video_file = self.target.path.join(self.target.working_directory,
+                                                       os.path.basename(self.host_video_file))
+        if self.force_dependency_push or not self.target.file_exists(self.device_video_file):
+            self.logger.debug('Copying {} to device.'.format(self.host_video_file))
+            self.target.push(self.host_video_file, self.device_video_file, timeout=120)
+
+        super(ExoPlayer, self).setup(context)
+
+        # Prepare logcat monitor
+        self.monitor = self.target.get_logcat_monitor(REGEXPS.values())
+        self.monitor.start()
+
+    def run(self, context):
+        play_cmd = 'am start -a {} -d "file://{}"'.format(self.action, self.device_video_file)
+        self.logger.info(play_cmd)
```

Comment:
This should not be an info-level log; probably does not need to be logged at all, as this will be logged as part of `execute()`

Reply:
Oops, yep this is debug leakage.